tests/Unit/Products/ProductImageTest.php

```diff
@@ -0,0 +1,148 @@
+    {
+        $productImage = $this->createProductImage();
+
+        $exists = $this->filesystem->disk('public')->exists($productImage->src);
+        $this->assertTrue($exists, 'The product image does NOT exist');
+    }
+
+    /** @test */
+    public function itThrowsAQueryExceptionWhenGivenAnInvalidProductId()
+    {
+        $this->expectException(QueryException::class);
+
+        $this->createProductImage([
+            'product_id' => 5,
+            'src' => 'somerandomstring'
+        ]);
+    }
+
+    /** @test */
+    public function itCanUpdateAProductImage()
+    {
+        $productImage = $this->createProductImage();
+
+        $newImage = UploadedFile::fake()->image('product.jpg', 600, 600);
+        $newFile = $this->storeFile($newImage);
+
+        $updated = $this->repo->update($productImage->id, [
+            'src' => $newFile
+        ]);
+
+        $this->assertTrue($updated, 'ProductImage did NOT update');
+        $this->assertDatabaseHas('product_images', ['src' => $newFile]);
+    }
+
+    /** @test */
+    public function itRemovesOldImageFromDiskWhenUpdatingImage()
+    {
+        $productImage = $this->createProductImage();
+        $oldFile = $productImage->src;
+
+        $newImage = UploadedFile::fake()->image('product.jpg', 600, 600);
+        $newFile = $this->storeFile($newImage);
+
+        $this->repo->update($productImage->id, [
+            'src' => $newFile
+        ]);
+
+        $oldFileExists = $this->filesystem->disk('public')->exists($oldFile);
+
+        $this->assertFalse($oldFileExists, 'ProductImage old file STILL exists');
+    }
+
+    /** @test */
+    public function itCanFailUpdatingAProductImage()
+    {
+        $updated = $this->repo->update(5, [
+            'src' => 'somerandomstring'
+        ]);
+
+        $this->assertFalse($updated, 'ProductImage DID update');
+    }
+
+    /** @test */
+    public function itRollsBackDatabaseAfterFailingToUpdateAProductImage()
+    {
+        $updated = $this->repo->update(5, [
+            'src' => 'somerandomstring'
+        ]);
+
+        $this->assertFalse($updated, 'ProductImage DID update');
+        $this->assertDatabaseMissing('product_images', ['src' => 'somerandomstring']);
+    }
+
+    /** @test */
+    public function itCanDeleteAProductImage()
+    {
+        $productImage = $this->createProductImage();
+        $deleted = $this->repo->delete($productImage->id);
+
+        $this->assertTrue($deleted);
+        $this->assertDatabaseMissing('product_images', ['src' => $productImage->src]);
+    }
+
+    /** @test */
+    public function itRemovesImageFromDiskWhenDeletingAProductImage()
+    {
+        $productImage = $this->createProductImage();
+        $this->repo->delete($productImage->id);
+
+        $exists = $this->filesystem->disk('public')->exists($productImage->src);
+        $this->assertFalse($exists, 'The product image still exists');
+    }
+
+    /** @test */
+    public function itCanFailDeletingAProductImage()
```

Comment:
As I noted regarding a similarly-named test, above, a better name here might be `itFailsToDeleteWhenProductImageIdIsInvalid()`.